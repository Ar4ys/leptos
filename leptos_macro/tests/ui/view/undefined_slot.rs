use leptos::*;

#[slot]
struct Then {}

#[slot]
struct ElseIf {
    #[prop(optional)]
    then: Option<Then>,
}

#[component]
fn If(then: Then) -> impl IntoView {
    _ = then;
}

#[component]
fn SlotIf(then: Then) -> impl IntoView {
    _ = then;
}

fn main() {
    view! {
        <If>
            <ElseIf slot />
        </If>
    };

    view! {
        <SlotIf>
            <ElseIf slot />
        </SlotIf>
    };
}
