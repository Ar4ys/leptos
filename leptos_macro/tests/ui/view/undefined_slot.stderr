error[E0599]: no method named `else_if` found for struct `IfPropsBuilder` in the current scope
  --> tests/ui/view/undefined_slot.rs:25:13
   |
12 |   #[component]
   |   ------------ method `else_if` not found for this struct
...
23 | /     view! {
24 | |         <If>
25 | |             <ElseIf slot />
   | |            -^^^^^^^^^^^^^^^ method not found in `IfPropsBuilder`
   | |____________|
   |

error[E0599]: no method named `else_if` found for struct `SlotIfPropsBuilder` in the current scope
  --> tests/ui/view/undefined_slot.rs:31:13
   |
17 |   #[component]
   |   ------------ method `else_if` not found for this struct
...
29 | /     view! {
30 | |         <SlotIf>
31 | |             <ElseIf slot />
   | |            -^^^^^^^^^^^^^^^ method not found in `SlotIfPropsBuilder`
   | |____________|
   |
